src/core/server/ui_settings/saved_objects/permission_control_ui_settings_wrapper.ts

```diff
@@ -0,0 +1,261 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  ACL,
+  SavedObject,
+  SavedObjectsBaseOptions,
+  SavedObjectsBulkCreateObject,
+  SavedObjectsBulkUpdateObject,
+  SavedObjectsBulkUpdateOptions,
+  SavedObjectsClientWrapperFactory,
+  SavedObjectsClientWrapperOptions,
+  SavedObjectsCreateOptions,
+  SavedObjectsErrorHelpers,
+  SavedObjectsUpdateOptions,
+} from 'opensearch-dashboards/server';
+import { getWorkspaceState } from 'opensearch-dashboards/server/utils';
+import { uiSettingWithPermission } from './ui_settings_permissions';
+
+const DASHBOARD_ADMIN_SETTINGS_ID = '_dashboard_admin';
+
+type AttributesObject = Record<string, unknown>;
+
+interface UiSettingWithPermission {
+  value: unknown;
+  isPermissionControlled: boolean;
+}
+
+interface ProcessedAttributes {
+  permissionAttributes: AttributesObject;
+  regularAttributes: AttributesObject;
+}
+
+export class PermissionControlUiSettingsWrapper {
+  private aclInstance?: ACL;
+
+  constructor(private readonly isPermissionControlEnabled: boolean) {}
+
+  public wrapperFactory: SavedObjectsClientWrapperFactory = (wrapperOptions) => {
+    const getUiSettingsWithPermission = async <T = unknown>(
+      type: string,
+      id: string,
+      options: SavedObjectsBaseOptions = {}
+    ): Promise<SavedObject<T>> => {
+      if (type === 'config') {
+        if (id === DASHBOARD_ADMIN_SETTINGS_ID) {
```

Comment:
When the id is not `DASHBOARD_ADMIN_SETTINGS_ID`, we should not change the behavior.